tests/integration/test_documentation_validation.py

```
@@ -247,28 +259,35 @@
+            s = re.sub(r'[^\w\s-]', '', s)
+            # Replace whitespace with single hyphen
+            s = re.sub(r'\s+', '-', s)
+            # Collapse multiple hyphens
+            s = re.sub(r'-{2,}', '-', s)
+            # Strip leading/trailing hyphens
+            s = s.strip('-')
+            return s
+
+        # Extract headers and internal links from the document
+        headers = [line.lstrip('#').strip() for line in summary_lines if line.startswith('#')]
+        valid_anchors: Set[str] = set(_to_gfm_anchor(header) for header in headers)
```

Ruff (0.14.6)

Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)